lib/chess/store/game.ex

```diff
@@ -22,20 +24,21 @@ defmodule Chess.Store.Game do
   """
   def create_changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, [:board, :user_id])
+    |> cast(params, [:board, :user_id, :opponent_id])
     |> put_change(:board, Board.default)
-    |> validate_required([:board, :user_id])
+    |> validate_required([:board, :user_id, :opponent_id])
   end
 
   def changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, [:board, :user_id])
-    |> validate_required([:board, :user_id])
+    |> cast(params, [:board, :user_id, :opponent_id])
+    |> validate_required([:board, :user_id, :opponent_id])
   end
 
-  def for_user(query, user) do
-    query
-    |> where([game], user_id: ^user.id)
+  def for_user(user) do
+    from game in Game,
+      where: game.user_id == ^user.id,
+      or_where: game.opponent_id == ^user.id
```

Comment:
heads up there's a big ol' asterisk when using `or_where` when using macros. You're using keyword expressions here which might not have the same behavior. tldr; if you chain `for_user` together with another expression afterwards, the later expression may be interpreted as `OR` statements instead of `AND` statements.